### PR TITLE
Remove the Jaeger Exporter

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-specification/issues/3980

Follows https://github.com/open-telemetry/opentelemetry-specification/issues/3551

Leftover after https://github.com/open-telemetry/opentelemetry-specification/pull/3567:

> The PR is now revised as follows:
> - the Jaeger Exporter is removed

### CODE REVIEW

**Comment by pellared** [comment]
@marcalff  Can you double-check please?

**Comment by marcalff** [comment]
> @marcalff Can you double-check please?

I tried to remove file `specification/trace/sdk_exporters/jaeger.md` before, but Yuri had concerns about preserving the content.

See https://github.com/open-telemetry/opentelemetry-specification/pull/3567#discussion_r1242479148

Please check with @yurishkuro

**Comment by pellared** [comment]
> See [#3567 (comment)](https://github.com/open-telemetry/opentelemetry-specification/pull/3567#discussion_r1242479148)

>> I don't think we should delete this, rather it should be moved somewhere collector-related.

1. AFAIK Collector has also removed this exporter. CC @open-telemetry/collector-approvers 
2. Having it as listed as a Tracing SDK exporter feels wrong.

**Comment by marcalff** [comment]
> > See [#3567 (comment)](https://github.com/open-telemetry/opentelemetry-specification/pull/3567#discussion_r1242479148)
> 
> > > I don't think we should delete this, rather it should be moved somewhere collector-related.
> 
>     1. AFAIK Collector has also removed this exporter. I guess the comment is outdated. CC @open-telemetry/collector-approvers
> 
>     2. Having it as listed as a Tracing SDK exporter feels wrong.

To clarify, I am fine with this cleanup, which I support, as the Jaeger exporter is no longer used.

However, I am not in a position to formally approve it, having no voice for specs.

File: specification/trace/sdk_exporters/jaeger.md [hunk 1]
```diff
@@ -1,166 +0,0 @@
-<!--- Hugo front matter used to generate the website version of this page:
-linkTitle: Jaeger
---->
-
-# OpenTelemetry to Jaeger Transformation
-
-**Status**: [Stable](../../document-status.md)
```
Comment (trask) [inline specification/trace/sdk_exporters/jaeger.md:7]: another option would be to keep the doc and mark it Deprecated

Comment (tigrannajaryan) [reply specification/trace/sdk_exporters/jaeger.md:7]: +1

**Review by yurishkuro — CHANGES_REQUESTED** [review]
So the situation today is that because in the past OTEL needed to transform to Jaeger format we wrote this transformation spec and there is a transformer in collector-contrib matching that spec. Even though the exporters in both SDK and collector may be safely removed now, the transformer is still actively used by Jaeger itself. So the simplest solution is to move the transformer spec to the transformer dir in collector-contrib. Longer term we can consider migrating that code completely to Jaeger, but the transformer is technically a public API in the contrib, we don't know if other people are also using it. 

Requesting changes.

**Comment by pellared** [comment]
@open-telemetry/collector-maintainers, what do you think of @yurishkuro proposal: https://github.com/open-telemetry/opentelemetry-specification/pull/3964#pullrequestreview-1960788192?

**Comment by pellared** [comment]
@open-telemetry/collector-contrib-maintainer, what do you think of @yurishkuro proposal: https://github.com/open-telemetry/opentelemetry-specification/pull/3964#pullrequestreview-1960788192? Are you fine moving the spec to collector-contrib? If so then I think this PR should blocked until the spec is added to collector.

**Comment by mx-psi** [comment]
Per our [CODEOWNERS](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/92d3e8e26335348b2607e529f29421f14b728057/.github/CODEOWNERS#L148) `pkg/translator/jaeger` is owned by @open-telemetry/collector-approvers and @frzifus. As part of said group I am in favor of moving the spec to collector-contrib for now, though personally I would like to see collector-contrib eventually be exclusively for components and move this library to some other repository

**Comment by yurishkuro** [comment]
I'd be fine with hosting the lib as a standalone repo in the jaeger org. At some point, if we convert the main jaeger repo into multi-module monorepo we could move it there.

**Comment by pellared** [comment]
@mx-psi I created https://github.com/open-telemetry/opentelemetry-specification/issues/3980 for tracking

**Comment by pellared** [comment]
Ready to review as https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32187 is already merged.

**Review by reyang — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Review by astencel-sumo — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Pull request merged**